src/pages/cart/index.tsx

```diff
@@ -1,5 +1,49 @@
-import { ReactElement } from 'react';
+import { ReactElement, useContext } from 'react';
+import { CartItem } from '../../components/general/cart-item';
+import { CartContext } from '../../contexts/Cart';
 
 export const CartPage = (): ReactElement => {
-  return <div>Cart Page</div>;
+  const { cart, setCart } = useContext(CartContext);
+
+  const increaseQuantity = (id: string) => {
+    setCart((currCart) => {
+      return currCart.map((currItem) => {
+        if (currItem.id === id) {
+          return { ...currItem, quantity: currItem.quantity + 1 };
+        }
+        return { ...currItem };
+      });
+    });
+  };
+  const decreaseQuantity = (id: string) => {
+    setCart((currCart) => {
+      return currCart.map((currItem) => {
+        if (currItem.id === id) {
+          return { ...currItem, quantity: currItem.quantity - 1 };
+        }
+        return { ...currItem };
+      });
+    });
+  };
+  const removeItem = (id: string) => {
+    setCart((currCart) => currCart.filter((cartItem) => id !== cartItem.id));
+  };
+  return (
+    <>
+    <h3>Cart Page</h3>
+      {cart.map((item) => {
+        return (
+          <CartItem
+            key={item.id}
+            imgSrc="https://picsum.photos/seed/test/270"
+            name={item.name}
+            quantity={item.quantity}
+            onDecreaseStepperClick={() => decreaseQuantity(item.id)}
```

Comment:
probably should have mentioned it on the call, but you can have just a stepper function and accept the amount to increment/decrement as a parameter as well. you can have +1 for increase and -1 for decrease as a parameter. it means you can have just one function - less code, more readable and less bugs :)

Reply:
Great idea - will do that now!